lib/nylas/registry.rb

```diff
@@ -0,0 +1,27 @@
+module Nylas
+  class Registry
```

Comment:
What is this registry for?

Reply:
Sure, this is a small implementation of the Registry pattern, we're using the Registry class in a couple places:

1. To register the different custom types that data gets serialized in and out of.
2. To register the attributes a given model has

We're using it to make sure that a reasonable error message is provided if we attempt to set or retrieve data that doesn't actually map accurately; as Ruby hashes don't have mechanisms for specifying that directly, and I didn't want to re-open classes at runtime.